### DISCUSSION TITLE
[HW1] Problem 4.3

### DISCUSSION THREAD

**Discussion by emjay73** [seominjoon/kaist-ai605#83] [Q&A] [answered]
I noticed that statement in problem 4.3  has changed from the initial version.

> Problem 4.3 (bonus) (2 points) Consider implementing bidirectional LSTM and two layers of LSTM.    
Concatenate the forward direction output at the final time step and the backward direction output at the first time step for the final classificaiton. Report your accuracy on dev data.    

In the modified statement, I'm not sure what the 'first timestep' means.
I thought bi-directional LSTM with text level information concatenates the last output (with last token) of the forward path and the last output (with first token) of the backward path.
So 'the backward direction output at the first time step' means the last output of the backward path?
Or is there any misunderstanding?

And I have one more question.
Do I have to implement both versions of bidirectional LSTM and two layers of LSTM?
Or just a single version with bidirectional LSTM + two layers of LSTM?

**Comment by seominjoon** [comment, marked as answer]
I modified the assignment to make it more clear. What I meant is that the last output of the backward path will be corresponding to the first time step in the forward direction, which I think you already guessed.

You can just implement a single version that has both bidirectional LSTM and two layers.

**Reply by emjay73** [reply]
1. So with regard to the time step, it is related to the original input word sequence, not LSTM sequence, right? I also do think I already guessed it correctly but still, the terminology is quite confusing for me. T-T. 
2. So it was meant to be a fused version..!  Thank you
